### PR TITLE
Update to Set Motion Sensor Accessory Type

### PR DESCRIPTION
**Pull request by jvmahon**
For consideration ...
I've modified the index.js to now allow for an explicit "MotionSensor" input accessory in addition to the "ContactSensor" accessory. 

The accessory type is selected by specifying "type":"ContactSensor" or "type":"MotionSensor" in the config.json file.  Additionally, for output accessories, the user can designate "type":"Switch".

This can probably be modified a bit more as you no longer really need the "mode" setting in index.js since the accessory type will now implicitly determine this.

### CODE REVIEW

**Comment by rsg98** [comment]
Thanks for the PR - will have a dig through as soon as I get chance.

File: index.js [hunk 1]
```diff
@@ -105,6 +105,17 @@ WPiPlatform.prototype.configureAccessory = function(accessory) {
       });
   }
 
+  if (accessory.getService(Service.MotionSensor) && accessory.context.mode === "in") {
```
Comment (rsg98) [inline index.js:108]: I'm thinking (as per your comment) that mode isn't really required any more - so if this service is defined and .type==="MotionSensor", then that should be the logic test - so .mode can be replaced with the .type check.

File: index.js [hunk 2]
```diff
@@ -143,11 +154,14 @@ WPiPlatform.prototype.addGPIOPin = function(gpiopin) {
       .setCharacteristic(Characteristic.Model, platform.config.model ? platform.config.model : "Pi GPIO")
       .setCharacteristic(Characteristic.SerialNumber, platform.config.serial ? platform.config.serial : "Default-SerialNumber");
 
-    switch(gpiopin.mode) {
-      case "out":
+    switch(true) {
```
Comment (rsg98) [inline index.js:192]: For this set of case statements, I'd prefer to preserve the original code, and do something like below (which probably contains a bunch of typos / bugs!)

```
// Use "type" logic to set services
if (typeof gpiopin.type !== 'undefined') {
 switch(gpiopin.type) {
   [your new case statement]
 }
} else {
 switch (gpiopin.mode) {
 [Original Case Statement]
}
```

I'm thinking this makes it easier to check that people without updated configuration files will continue to work as anticipated - but your PR also lays good ground work I need to build into the next version of this module - on the version 2 alpha branch currently.

Hope that all makes sense!

**Review by rsg98 — CHANGES_REQUESTED** [review]

**Comment by jvmahon** [comment]
I'll defer to you on how to incorporate the changes back into the Master branch. As a further suggestion, there are other service types which may also make sense for the GPIO pin service types, e.g., "LeakSensor", "OccupancySensor", "SmokeSensor", etc. which might be likewise be easily integrated into the code using the "type" designation.

**Comment by jvmahon** [comment]
I've further updated the index.js and config.json to add additional service types "LeakSensor", "OccupancySensor", "SmokeSensor".

File: index.js (hunk 2 above)
Comment (jvmahon) [reply index.js:192]: I leave this to you, but note I've further updated to allow additional device types. Also, though the gpio pin "mode" setting is now implicit in the device type and "mode" could be eliminated, I haven't looked into how the gpio pin settings function works;  so, though the pin "mode" should be  redundant, its left in for now for gpio pin settings purposes. "Mode" should be eliminated and the gpio pin settings determined by the type so you don't have inconsistencies between the device type and the pin settings, but I haven't figured that out yet. Perhaps you can -- or I may try some more experimenting at a later date.

**Comment by Kimovitzh** [comment]
Is it possible to include the LockMechanism?

**Comment by jvmahon** [comment]
@Kimovitzh  - I don't understand - how would you be using the LockMechanism service with a GPIO pin?

**Comment by Kimovitzh** [comment]
@rsg98 with one of these https://www.fotoagent.dk/single_picture/11708/138/mega/M-622701712.jpg
I'm current running my lock with HAP-NodeJS but what to use the homebridge.

**Comment by jvmahon** [comment]
Can you provide a bit more detail - what kind of lock are you using and how are you currently connecting it to HAP-NodeJS?

Thanks

**Comment by rsg98** [comment]
If you want to read the lock state on a separate pin to setting it, then
not yet - that’s similar functionality to the stateful programmable Switch
support that’s needed.

If you trust the gpio pin state as being equivalent to the lock state, then
(at present) you would need to edit the plugin code to replace the switch
accessory class with Lock.

I’m planning on allowing better customisation of accessory types in the
next version, but haven’t had any time to work on it recently.

**Comment by Kimovitzh** [comment]
@rsg98 I'm using a ULN2003 to lock and unlock a electric lock, and just want to unlock and lock my door using homebridge.
It looks as if [this](https://github.com/dubocr/homebridge-gpio-device) can do it.

**Comment by jvmahon** [comment]
Unfortunately, with that method of driving a lock, you’re now in the “beyond my understanding” area for this plugin. I only added some simple type additions in my PR.

**Comment by jvmahon** [comment]
Thakanks for merging in those changes. I just noticed I had not updated the package # in package.json. I assume you'll do that before publishing on npm.

Best regards.

**Pull request merged**